diploma-project/components/TrendingCard.jsx

```diff
@@ -1,16 +1,23 @@
 import Image from "next/future/image";
+import {useRouter} from "next/router";
+
+const TrendingCard = ({product}) => {
+    const router = useRouter();
+
+    const onClick = () => {
+        void router.push(`/product/${product.id}`);
+    };
 
-const TrendingCard = () => {
     return (
         <article className="card sm:card-side bg-base-100 border-b-2 sm:border-b-0 border-gray-400 w-full h-full flex">
-            <figure className="relative h-72 sm:h-96 sm:w-96">
-                <Image src="/recommended_ring.jpg" alt="Trending ring" className="object-cover object-center" fill />
+            <figure className="relative h-72 sm:w-full">
+                <Image src={product.displayImageURL} alt="Trending ring" className="object-cover object-center" fill />
             </figure>
             <div className="card-body">
-                <h2 className="card-title">Trending and popular!</h2>
-                <p>Click the button to see this masterpiece.</p>
+                <h2 className="card-title">{product.title}</h2>
+                <p className="sm:pr-10">{product.description}</p>
                 <div className="card-actions justify-end">
-                    <button className="btn btn-primary">More</button>
+                    <button className="btn btn-primary" onClick={onClick}>More</button>
```

Comment:
This is an example of what I mentioned - the onClick function navigates to a new page, but with the current implementation SEO won't work. If the onClick is used only to navigate to another page (like it's here), you should use Link